docs/design/plugin-registry.md

```diff
@@ -0,0 +1,105 @@
+  - @shubham-pampattiwar
+  - @eemcmullan
+  - @savitharaghunathan
+  - @dymurray
+  - @shawn-hurley
+approvers:
+  - @dymurray
+  - @shawn-hurley
+  - @shubham-pampattiwar
+creation-date: 2022-06-14
+last-updated: 2022-06-14
+status: implementable
+---
+
+# Plugin Registry
+
+Plugin registry enhancement replaces current usage of [registry](https://github.com/konveyor/distribution/) [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) insecure [routes](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html/networking/configuring-routes)
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+## Open Questions [optional]
+
+Do we want registry logs to be included in velero backup and/or restore logs?
+
+## Summary
+
+removes a listening HTTP server and connect [openshift-velero-plugin](https://github.com/openshift/openshift-velero-plugin/) directly with the [distribution/distribution](https://github.com/distribution/distribution) code to serve requests from [container/image/copy](https://github.com/containers/image/tree/main/copy) method as needed. A "library" to help achieve this is [kaovilai/udistribution](https://github.com/kaovilai/udistribution) which may be moved to konveyor org.
+
+## Motivation
+
+Current image backup workflow depends on registry deployment with exposed insecure routes allowing anyone knowing the URL of the exposed routes to push or pull images to the registry.
+
+### Goals
+
+Exposed routes are eliminated.
+
+### Non-Goals
+
+Creating another custom resource and a controller to replace usage of [distribution/distribution](https://github.com/distribution/distribution).
+
+## Proposal
+
+We propose that insecure routes are eliminated by removing registry deployment, and making network requests to [distribution/distribution](https://github.com/distribution/distribution) storage drivers such as [s3-aws](https://github.com/distribution/distribution/blob/main/registry/storage/driver/s3-aws/s3.go) from within velero pod via openshift-velero-plugin.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The logs relating to registry will be mixed into velero pod logs
+
+### Security, Risks, and Mitigations
+
+As more backups and/or new Backup Storage Location are added and thus caching more initialized registries configurations, we may want to periodically garbage collect unused or outdated backup storage location registry configurations to optimize memory usage.
+
+## Design Details
+
+OADP-Operator controller manager will
+ * set environment variables on the velero container
+   * tell openshift-velero-plugin to initialize [udistribution transport](https://github.com/kaovilai/udistribution/blob/main/pkg/image/udistribution/docker_transport.go#L36) to be used to talk to storage drivers such as s3
```

Comment:
will be adding lucidchart to https://github.com/kaovilai/udistribution/ and add a summary here of what udistribution repo contains.

Reply:
@kaovilai thank you, yes this will be very helpful to understand the workflow.

Reply:
This is the published lucidchart. It will be added to PR soon.
https://lucid.app/documents/view/14dd9d7d-38a5-40f7-a378-c89a99a1ace6